pkg/server/api/entry/v1/service_test.go

```diff
@@ -2,29 +2,246 @@ package entry_test
 
 import (
 	"context"
+	"errors"
 	"testing"
+	"time"
 
+	"github.com/sirupsen/logrus"
 	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/server/api"
 	"github.com/spiffe/spire/pkg/server/api/entry/v1"
 	"github.com/spiffe/spire/pkg/server/api/rpccontext"
 	entrypb "github.com/spiffe/spire/proto/spire-next/api/server/entry/v1"
+	"github.com/spiffe/spire/proto/spire-next/types"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/server/datastore"
 	"github.com/spiffe/spire/test/fakes/fakedatastore"
 	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 )
 
-type serviceTest struct { //nolint: unused,deadcode
+var (
+	ctx         = context.Background()
+	td          = spiffeid.RequireTrustDomainFromString("example.org")
+	federatedTd = spiffeid.RequireTrustDomainFromString("domain1.org")
+)
+
+func TestGetEntry(t *testing.T) {
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	test.createBundles(t)
```

Comment:
Can we add a comment here that the reason we need to create a bundle is so that the entry can "federate with" that bundle?